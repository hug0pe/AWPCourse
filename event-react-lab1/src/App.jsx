import reactLogo from "./assets/react.svg";
import viteLogo from "/vite.svg";
import { Counter } from "./components/Counter";
import { AddNumber } from "./components/AddNumber";
import { InputText } from "./components/InputText";

import "./App.css";

function App() {
  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="" />
        </a>
        <a href="https://reactjs.org" target="_blank">
          <img src={reactLogo} className="logo react" alt="" />
        </a>
      </div>
      <h1>Vite + React</h1>

      <Counter />
      <InputText />
      <AddNumber />

      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </div>
  );
}

export default App;
