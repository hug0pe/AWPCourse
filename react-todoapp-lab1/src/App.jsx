import React, { useState } from "react";
import Todos from "./components/Todos";
import EditTodo from "./components/EditTodo";
import { useRoutes, Link, useQueryParams } from "raviger";

const routes = {
  '/': () => <Todos />,
  '/edit/:id': ({id}) => <EditTodo id={id} />,
}

function App() {
  const [query, setQuery] = useQueryParams();
  const route = useRoutes(routes, query);

  return <>{route}</>;
}

export default App;