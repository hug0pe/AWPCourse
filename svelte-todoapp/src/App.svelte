<script>
  import svelteLogo from "./assets/svelte.svg";
  import viteLogo from "/vite.svg";
  import Counter from "./components/Counter.svelte";
  import List from "./components/List.svelte";
  import AddTask from "./components/AddTask.svelte";

  /**
   * [
      { text: 'Learn js', done: false },
      { text: 'Clean bedroom', done: true },
      { text: 'TODO ??', done: false },
    ]
   */
  let tasks = [
      { text: 'Learn js', done: false },
      { text: 'Clean bedroom', done: true },
      { text: 'TODO ??', done: false },
    ];

  const changeDone = (index) => {
    const newTasks = [...tasks]; // copie du tableau
    newTasks[index] = { ...newTasks[index], done: !newTasks[index].done };

    tasks =newTasks;
  };

  const addTask = (text) => {
    const task = {
      text: text,
      done: false,
    };
    tasks = [...tasks, task];
  };
</script>

<main>
  <div>
    <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
      <img src={viteLogo} class="logo" alt="Vite Logo" />
    </a>
    <a href="https://svelte.dev" target="_blank" rel="noreferrer">
      <img src={svelteLogo} class="logo svelte" alt="Svelte Logo" />
    </a>
  </div>
  <h1>Vite + Svelte</h1>

  <div class="card">
    <Counter />
  </div>

  <p>
    Check out <a
      href="https://github.com/sveltejs/kit#readme"
      target="_blank"
      rel="noreferrer">SvelteKit</a
    >, the official Svelte app framework powered by Vite!
  </p>

  <p class="read-the-docs">Click on the Vite and Svelte logos to learn more</p>

  <div class="todo-app">
    <h1>Todo List App</h1>

    <!--Todo -->
    <List tasks={tasks} changeDone={changeDone}/>
     <!--Todo -->
    <AddTask addTask={addTask}/>
  </div>
</main>

<style>
  .logo {
    height: 6em;
    padding: 1.5em;
    will-change: filter;
    transition: filter 300ms;
  }
  .logo:hover {
    filter: drop-shadow(0 0 2em #646cffaa);
  }
  .logo.svelte:hover {
    filter: drop-shadow(0 0 2em #ff3e00aa);
  }
  .read-the-docs {
    color: #888;
  }
  .todo-app {
    text-align: left;
  }
</style>