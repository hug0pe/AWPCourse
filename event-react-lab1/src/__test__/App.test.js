import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";

import "@testing-library/jest-dom";

import App from "../App";

describe("App should render", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the Vite and React logos", () => {
    render(<App />);

    expect(screen.getByAltText("Vite logo")).toBeInTheDocument();
    expect(screen.getByAltText("React logo")).toBeInTheDocument();
  });

  it('renders the "Click on the Vite and React logos to learn more" text', () => {
    render(<App />);

    expect(
      screen.getByText(/Click on the Vite and React logos to learn more/i)
    ).toBeInTheDocument();
  });
});
