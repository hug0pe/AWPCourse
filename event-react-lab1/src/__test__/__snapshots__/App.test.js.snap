// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        className="logo"
        src="test-file-stub"
      />
    </a>
    <a
      href="https://reactjs.org"
      target="_blank"
    >
      <img
        alt="React logo"
        className="logo react"
        src="test-file-stub"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    className="card"
  >
    <button
      onClick={[Function]}
    >
      count is 
      0
    </button>
  </div>
  <div
    className="card"
  >
    <p>
      <input
        onChange={[Function]}
        type="text"
      />
       
      <br />
      <strong>
        state input
      </strong>
      : 
    </p>
  </div>
  <div
    className="card"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <input
        onChange={[Function]}
        onClick={[Function]}
        type="text"
      />
       
      Add number 
      0
    </button>
  </div>
  <p
    className="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
